packages/core/src/sync-historical/index.ts

```diff
@@ -465,11 +492,30 @@ export const createHistoricalSync = async (
 
     if (isKilled) return;
 
-    const transactionReceipts = await Promise.all(
-      Array.from(transactionHashes).map((hash) =>
-        _eth_getTransactionReceipt(args.requestQueue, { hash }),
-      ),
-    );
+    const blockReceipts = (await Promise.all(
+      [...requiredBlockReceipts].map((blockNumber) => _eth_getBlockReceipts(args.requestQueue, {
+        blockNumber,
+      }).then(receipts => ({ blockNumber, receipts })))
+    ));
```

Comment:
Curious if you think block receipts would benefit from using the same "cache" concept that is used in this file for block and traces?

Reply:
I think it would definitely benefit. Included in the commit below.